int32 id
string message
float64 value
bool flag 